File: Lab_2/task_1/1.cpp
#include <iostream>
#include <string>

using namespace std;

string compressString(const string& s) {
    string compressed;
    int count = 1;

    for (size_t i = 1; i <= s.length(); ++i) {
        if (i < s.length() && s[i] == s[i - 1]) {
            ++count;
        } else {
            compressed += s[i - 1] + to_string(count);
            count = 1;
        }
    }

    if (compressed.length() < s.length()) {
        return compressed;
    } else {
        cout << "Сжатие не уменьшило строку. Возвращена исходная строка." << endl;
        return s;
    }
}

int main() {
    string s;
    cout << "Введите строку: ";
    cin >> s;
    cout << "Результат: " << compressString(s) << endl;
    return 0;
}

File: Lab_2/task_1/1.java
import java.util.Scanner;
import java.io.PrintStream;  import static java.nio.charset.StandardCharsets.UTF_8;

public class StringCompression {
    public static String compressString(String s) {
        PrintStream out = new PrintStream(System.out, true, UTF_8);
        StringBuilder compressed = new StringBuilder();
        int count = 1;

        for (int i = 1; i <= s.length(); i++) {
            if (i < s.length() && s.charAt(i) == s.charAt(i - 1)) {
                count++;
            } else {
                compressed.append(s.charAt(i - 1)).append(count);
                count = 1;
            }
        }

        if (compressed.length() < s.length()) {
            return compressed.toString();
        } else {
            out.println("Сжатие не уменьшило строку. Возвращена исходная строка.");
            return s;
        }
    }

    public static void main(String[] args) {
        PrintStream out = new PrintStream(System.out, true, UTF_8);
        Scanner scanner = new Scanner(System.in);
        out.print("Введите строку: ");
        String s = scanner.nextLine();
        out.println("Результат: " + compressString(s));
        scanner.close();
    }
}
